tests/unit/browser/test_adblock.py

```diff
@@ -0,0 +1,434 @@
+# TODO See utils/test_standarddirutils for OSEError and caplog assertion
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+#!/usr/bin/env python3
+
+# Copyright 2014-2015 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutebrowser.browser.adblock."""
+
+import os
+import zipfile
+import shutil
+
+import pytest
+
+from PyQt5.QtCore import pyqtSignal, QUrl, QObject
+
+from qutebrowser.browser import adblock
+from qutebrowser.utils import objreg
+from qutebrowser.commands import cmdexc
+
+
+WHITELISTED_HOSTS = ('qutebrowser.org', 'mediumhost.io')
+
+BLOCKLIST_HOSTS = ('localhost',
+                   'mediumhost.io',
+                   'malware.badhost.org',
+                   '4-verybadhost.com',
+                   'ads.worsthostever.net')
+
+CLEAN_HOSTS = ('goodhost.gov', 'verygoodhost.com')
+
+URLS_TO_CHECK = ('http://localhost',
+                 'http://mediumhost.io',
+                 'http://malware.badhost.org',
+                 'http://4-verybadhost.com',
+                 'http://ads.worsthostever.net',
+                 'http://goodhost.gov',
+                 'ftp://verygoodhost.com'
+                 'http://qutebrowser.org')
+
+
+@pytest.fixture
+def data_tmpdir(monkeypatch, tmpdir):
+
+    """Set tmpdir as datadir"""
+
+    tmpdir = str(tmpdir)
+    monkeypatch.setattr('qutebrowser.utils.standarddir.data', lambda: tmpdir)
+
+
+# XXX Why does read_hosts needs basedir to be None
+# in order to print message 'run adblock-update to read host blocklist' ?
```

Comment:
Ok I get it ... then shouldn't it be 
'if args.temp_basedir is False' instead of 'if args.basedir is None' on line 138 of adblock.py ?
That way someone who would specify a custom basedir as a command line argument would still be prompted to update.